Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/COREX12AuditTransforms.java

```diff
@@ -0,0 +1,223 @@
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the documentation
+ *       and/or other materials provided with the distribution.
+ *     * Neither the name of the United States Government nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.corex12.docsubmission.audit.transform;
+
+import com.services.nhinc.schema.auditmessage.AuditMessageType;
+import com.services.nhinc.schema.auditmessage.ParticipantObjectIdentificationType;
+import com.services.nhinc.schema.auditmessage.TypeValuePairType;
+import gov.hhs.fha.nhinc.audit.transform.AuditTransforms;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.corex12.docsubmission.audit.CORE_X12AuditDataTransformConstants;
+import gov.hhs.fha.nhinc.transform.marshallers.JAXBContextHandler;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.namespace.QName;
+import org.apache.log4j.Logger;
+
+/**
+ * An abstract class which implements methods common across CORE X12 RealTime and BatchRequest and Response services.
+ *
+ * @author achidamb
+ * @param <T>
+ * @param <K>
+ */
+public abstract class COREX12AuditTransforms<T, K> extends AuditTransforms<T, K> {
+
+    private static final Logger LOG = Logger.getLogger(COREX12AuditTransforms.class);
+
+    /**
+     * Build auditMsg with ParticipantObjectIdentificationType
+     *
+     * @param request
+     * @param assertion
+     * @param auditMsg
+     * @return
+     */
+    @Override
+    protected final AuditMessageType getParticipantObjectIdentificationForRequest(T request,
+        AssertionType assertion, AuditMessageType auditMsg) {
+
+        ParticipantObjectIdentificationType participantObject
+            = buildBaseParticipantObjectIdentificationType(getPayloadFromRequest(request));
+        byte[] baos = marshallToByteArrayFromRequest(request);
```

Comment:
You don't need to change it for this PR, but for the future, baos is typically used as a name for a ByteArrayOutputStream, not a byte[]